cmd/adapter/config_test.go

```diff
@@ -0,0 +1,461 @@
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/gravitational/logging-app/cmd/adapter/k8s"
+	"github.com/logrange/range/pkg/transport"
+)
+
+func TestLoadCfgFromFile(t *testing.T) {
+	t.Run("test1", func(t *testing.T) {
+		// write
+		cfg := NewDefaultConfig()
+		bb, _ := json.Marshal(cfg)
+		fd, _ := writeTmpJsonFile(bb)
+		defer os.Remove(fd.Name())
+
+		// read
+		got, err := LoadCfgFromFile(fd.Name())
+		if err != nil {
+			t.Errorf("LoadCfgFromFile() error = %v", err)
+			return
+		}
+
+		// check
+		if !reflect.DeepEqual(got, cfg) {
+			t.Errorf("LoadCfgFromFile() = %v, want %v", got, cfg)
+		}
+	})
+}
+
+func TestConfig_Merge(t *testing.T) {
+	type fields struct {
+		Gravity         *gravity
+		Logrange        *logrange
+		SyncIntervalSec int
+	}
+	type args struct {
+		other *Config
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   *Config
+	}{
+		{name: "test1",
+			fields: fields{
+				Gravity:         newDefaultGravityConfig(),
+				Logrange:        newDefaultLograngeConfig(),
+				SyncIntervalSec: 123,
+			},
+			args: args{
+				other: &Config{
+					Gravity: &gravity{
+						ApiListenAddr: "127.0.0.123:1234",
+						Kubernetes:    &k8s.Config{Namespace: "gravity"},
+					},
+					Logrange: &logrange{
+						Partition:         "partition",
+						Kubernetes:        &k8s.Config{Namespace: "logrange"},
+						ForwarderTmplFile: "file1",
+						Transport:         &transport.Config{Tls2Way: utils.BoolPtr(true)},
+					},
+				},
+			},
+			want: &Config{
+				Gravity: &gravity{
+					ApiListenAddr: "127.0.0.123:1234",
+					Kubernetes: &k8s.Config{
+						Namespace:              "gravity",
+						ForwarderConfigMapName: "log-forwarders",
+					},
+				},
+				Logrange: &logrange{
+					Partition:         "partition",
+					ForwarderTmplFile: "file1",
+					Kubernetes: &k8s.Config{
+						Namespace:              "logrange",
+						ForwarderConfigMapName: "lr-forwarder",
+					},
+					Transport: &transport.Config{
+						ListenAddr: "127.0.0.1:9966",
+						Tls2Way:    utils.BoolPtr(true),
+					},
+				},
+				SyncIntervalSec: 123,
+			}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Config{
+				Gravity:         tt.fields.Gravity,
+				Logrange:        tt.fields.Logrange,
+				SyncIntervalSec: tt.fields.SyncIntervalSec,
+			}
+			c.Merge(tt.args.other)
+			if !reflect.DeepEqual(c, tt.want) {
+				t.Errorf("Config.Merge() = %v, want %v", c, tt.want)
+			}
+		})
+	}
+}
+
+func TestConfig_Check(t *testing.T) {
+	type fields struct {
+		Gravity         *gravity
+		Logrange        *logrange
+		SyncIntervalSec int
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr error
+	}{
+		{name: "test1",
+			fields: fields{
+				Gravity:         newDefaultGravityConfig(),
+				Logrange:        newDefaultLograngeConfig(),
+				SyncIntervalSec: 123,
+			},
+			wantErr: nil,
+		},
+
+		{name: "test2",
```

Comment:
For example, instead of the hollow `test2`, I'd write `receives error if gravity config is empty`

Reply:
applied